packages/formation-react/src/components/AlertBox/AlertBox.jsx

```diff
@@ -32,13 +36,14 @@ class AlertBox extends React.Component {
       return;
     }
 
-    const isInView = window.scrollY <= this._ref.offsetTop;
-
-    if (this._ref && !isInView) {
-      this._ref.scrollIntoView({
-        block: this.props.scrollPosition,
-        behavior: 'smooth',
-      });
+    if (this._ref) {
+      clearTimeout(this.scrollToAlertTimeout);
+      this.scrollToAlertTimeout = setTimeout(() => {
+        this._ref.scrollIntoView({
+          block: this.props.scrollPosition,
+          behavior: 'smooth',
+        });
+      }, 0);
```

Comment:
> I needed to use a `setTimeout` to ensure the ref in the scroll logic was defined

Can you elaborate on this? I apologize if I'm missing something obvious - it just isn't clear why we have `setTimeout` with a delay of 0 along with a conditional check for the ref.

Would any of this be simplified if this component could use a [hook-based ref](https://reactjs.org/docs/hooks-reference.html#useref)?

Reply:
@bkjohnson I needed the `ref` to get in the event loop because it wouldn't point to the DOM node otherwise - since it's not painted yet. I believe using a hook-based ref would result in more or less the same logic 🙂

Reply:
React's [docs on `refs`](https://reactjs.org/docs/refs-and-the-dom.html#callback-refs) say that:

> Refs are guaranteed to be up-to-date before `componentDidMount` or `componentDidUpdate` fires.

Is there still something missing?

Reply:
Correct, I read that, too. However, if you try it without wrapping it in a setTimeout, [this line](https://github.com/department-of-veterans-affairs/veteran-facing-services-tools/pull/452/files#diff-4c7206ecbf59396b33d877282274b633R39) equates to false because `this._ref` is undefined. I have also experimented with using `React.createRef` and the same issue occurs where the ref isn't set yet when `scrollToAlert` is called within componentDidMount. Feel free to pull down the branch and you will see what I mean

Reply:
I used `yarn link` to connect this branch to `vets-website`, and it doesn't look like `this._ref` is undefined in any of the unit tests:

![image](https://user-images.githubusercontent.com/2008881/95238410-55d2ef80-07be-11eb-8157-2d7de4297839.png)


![image](https://user-images.githubusercontent.com/2008881/95238504-769b4500-07be-11eb-9061-fac2c270c75f.png)

Can you show me where it is appearing as undefined?

Reply:
I took a look at this locally as well. And what I found was that checking for `this._ref` in `scrollToAlert` was not required, but putting the `scrollIntoView()` call in a timeout was. This GIF shows the Alert not scrolling into view when the `scrollIntoView()` call is outside of a timeout and then it successfully scrolling into view when the call is made in a timeout. I'm very perplexed why this is needed.

![without and with timeout](https://user-images.githubusercontent.com/20728956/95276299-ac145280-07ff-11eb-88d3-07c673690b62.gif)

<img width="681" alt="Screen Shot 2020-10-06 at 3 40 54 PM" src="https://user-images.githubusercontent.com/20728956/95276317-bcc4c880-07ff-11eb-9ed7-cb1b06c36b09.png">

In summary: the `this_ref` check does not seem needed. But delaying the `scrollIntoView()` call until the next run of the event loop seems to be needed 🤔

Reply:
So @erikphansen and I were testing this and it looks like I was mistaken. `this._ref` is not `undefined`, but it does not seem to scroll without putting it in the event loop (likely due to the dom node not being fully painted at the point of execution, just a hypothesis though). Here is a screenshot from Erik:

![image (9)](https://user-images.githubusercontent.com/14869324/95277182-6f018d80-080a-11eb-9d01-00374365f0b4.png)

Reply:
This is strange, but I think it's ok for now since an alternative would likely involve some out of scope refactoring. Can you add some comments around the `clearTimeout` and `setTimeout` calls to explain why they are there?

Reply:
Amen on this being strange. But I'm with Sandra. I don't think React has actually added the element to the DOM when it calls `scrollIntoView()`.

Reply:
I also see no reason to store the timeout and clean it up since it essentially executes immediately.

Reply:
I try to follow the best practices when it comes to using `setTimeout` and `setInterval` to always clear them in order to prevent memory leaks. In this scenario, the probability is very low (if not 0) due to the setTimeout time being set to 0, but I like to play it safe anyways in case someone were to change that number in the future

Reply:
@bkjohnson I added a comment to the setTimeout, good idea!